File: serene/src/Serene.Web/Modules/ServerTypes/Workflow/DocumentSubmitForm.ts
﻿import { LookupEditor, TextAreaEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface DocumentSubmitForm {
    Language: LookupEditor;
    Comment: TextAreaEditor;
}

export class DocumentSubmitForm extends PrefixedContext {
    static readonly formKey = 'Workflow.DocumentSubmit';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!DocumentSubmitForm.init)  {
            DocumentSubmitForm.init = true;

            var w0 = LookupEditor;
            var w1 = TextAreaEditor;

            initFormType(DocumentSubmitForm, [
                'Language', w0,
                'Comment', w1
            ]);
        }
    }
}

File: serene/src/Serene.Web/Modules/Workflow/Forms/DocumentSubmitForm.cs
using Serene.Administration;

namespace Serene.Workflow.Forms;

[FormScript("Workflow.DocumentSubmit")]
public class DocumentSubmitForm
{
    [LookupEditor(typeof(LanguageRow))]
    public int Language { get; set; }
    [TextAreaEditor(Rows = 3)]
    public string Comment { get; set; }
}
